manifests/agent.pp

```diff
@@ -244,37 +244,28 @@
   }
 
   ## ossec.conf generation concats
-  case $::kernel {
-    'Linux': {
-      case $::osfamily {
-        'Redhat', 'redhat', 'OracleLinux': {
-          $apply_template_os = 'rhel'
-          if ( $::operatingsystemrelease =~ /^7.*/ ) {
-            $rhel_version = '7'
-          } elsif ( $::operatingsystemrelease =~ /^6.*/ ) {
-            $rhel_version = '6'
-          } elsif ( $::operatingsystemrelease =~ /^5.*/ ) {
-            $rhel_version = '5'
-          } else {
-            fail('This ossec module has not been tested on your distribution')
-          }
-        }
-        'Debian', 'debian', 'Ubuntu', 'ubuntu': {
-          $apply_template_os = 'debian'
-          if ( $::lsbdistcodename == 'wheezy') or ($::lsbdistcodename == 'jessie') {
-            $debian_additional_templates = 'yes'
-          }
-        }
-        'Amazon': {
-          $apply_template_os = 'amazon'
-        }
-        'CentOS', 'Centos', 'centos': {
-          $apply_template_os = 'centos'
-        }
-        default: { fail('This ossec module has not been tested on your distribution') }
+  case $::operatingsystem{
```

Comment:
I prefer the  `case $::kernel {` block, we use it multiple times in the file instead of just filtering by `operatingssystem` which raises few questions like:
- All Windows do have the same `operatingsystem`? (Windows server, Windows 10, etc...)
- Is `windows` and not `Windows` the proper operatingsystem?

Please review best practices and or other common manifests examples and let me know your conclusions and required changes.

Thanks!

Reply:
Hi @jm404 

I agree with you, dealing with _Windows_ case by only checking the _operating system_ could produce errors.

To solve such an issue, I added a higher layer of a conditional statement check, which checks the Kernel first, if It's _Linux_, then it proceeds to run the _operating system_ case condition. If it's _Windows_ then it's directly select the corresponding config.

Find the changes in https://github.com/wazuh/wazuh-puppet/pull/197/commits/b1a706f5f6d4be3e816bd561a432e541b3c9eefa.

Kr,

Rsjad